scripts/zk_benchmark.py

```diff
@@ -0,0 +1,290 @@
+        rem_size = child_node_config.number
+        result = []
+        while rem_size > 0:
+            gen_size = min(rem_size, self.config.chunk_size)
+            result.append(ChildNodeConfig(
+                gen_size, 
+                fixed_name=child_node_config.fixed_name, 
+                name_len=child_node_config.name_len, 
+                data_bytes=child_node_config.data_bytes))
+            rem_size -= gen_size
+        
+        return result
+
+    def load_data(self, parent: str, child_node_config: ChildNodeConfig):
+        configs = self.__get_chunked_configs(child_node_config)
+        with concurrent.futures.ProcessPoolExecutor(max_workers=self.config.parallelism) as exec:
+            futures_to_id = {exec.submit(self._process, parent, config): id for (id, config) in enumerate(configs)}
+            for future in concurrent.futures.as_completed(futures_to_id):
+                id = futures_to_id[future]
+                try:
+                    res = future.result()
+                except Exception as exc:
+                    print("process: {} generated an exception: {}".format(id, exc))
+
+    def _process(self, parent: str, child_node_config: ChildNodeConfig):
+        with DataLoader(self.config, zk_config=self.zk_config) as loader:
+            loader.load_data(parent, child_node_config)
+
+class Benchmark:
+    def __init__(self, config: BenchmarkConfig):
+        self.config = config
+        self.zk_config = config.zk_config
+        self.data_loader_config = config.data_loader_config
+        self.root_node = config.root_node
+    
+    def run(self, skip_measure: bool = False, use_multiprocess: bool = False):
+        for (idx, run) in enumerate(self.config.runs):
+            print("Starting benchmark run: {} [{} of {}]".format(run.name, idx+1, len(self.config.runs)))
+            self._run(run, skip_measure, use_multiprocess)
+    
+    def cleanup(self):
+        with ZkClientWrapper(config=self.zk_config, name="cleanup_zk") as client:
+            client.clean_all()
+    
+    def _run(self, run_config: BenchmarkRunConfig, skip_measure: bool, use_multiprocess: bool):
+        path = self._get_parent_path(run_config.name)
+        measure_samples = 5
+
+        ## DataLoading step
+        print("Loading nodes under path:", path)
+        with self._get_data_loader(self.data_loader_config, self.zk_config, use_multiprocess) as loader:
+            loader.load_data(run_config.name, run_config.child_node_config)
+
+        ## Measure step
+        if not skip_measure:
+            with ZkClientWrapper(config=self.zk_config, name="measure_zk") as client:
+                print("Num child nodes under path {}: {}".format(path, len(client.ls(path))))
+                with catchtime() as t:
+                    for _ in range(measure_samples):
+                        client.ls(path)
+                print("Latency get_children on path {}: {} ms".format(path, t.time / measure_samples))
```

Comment:
in this case it wouldn't matter much, but it would be good to have it as min, max, avg  over the sample range.